### PR TITLE
initial 8-bit ascii selector support

### PR DESCRIPTION
**Pull request by stephen-hawley**
What's going on:
We use UTF-16 selectors which are 16-bit characters. ObjC uses C string selectors which are 8 bit null terminated ascii.
16 bit selectors will use 2x the storage and for each invoke to `sel_registerName` which cause a new string to be allocated and the input string will be converted to 8-bit ascii. We can do better.

This change represents a proof of concept of using 8 bit selectors only. How?

- Implemented an overload of `Selector.GetHandle` which takes a `ReadOnlySpan<byte>`
- Added a fake overload of the `GetHandle` pinvoke that takes a `byte*` instead of `string`.
- Changed `UIControl` to use this overload.
- To generate the selector in `UIControl`, I used this code:
```
if (args.Length != 2) {
        Console.WriteLine ("Usage: vert varname string");
        return;
}

var varname = args [0];
var str = args [1];

Console.WriteLine ($"static byte [] {varname} = new byte [] {{");
for (var i=0; i < str.Count (); i++) {
        if (i > 0) Console.Write (',');
        if (i % 8 == 0) {
                Console.Write ("\n\t");
        } else {
                Console.Write (' ');
        }
        Console.Write ($"(byte)'{str [i]}'");
}
Console.WriteLine (", 0\n};");
```
and finally, for testing, I ran this code:
```
public override bool FinishedLaunching (UIApplication application, NSDictionary launchOptions)
        {
                // create a new window instance based on the screen size
                Window = new UIWindow (UIScreen.MainScreen.Bounds);

                // create a UIViewController with a single UILabel
                var vc = new UIViewController ();

                actionB = new UIButton (new CGRect (0, 0, 300, 150));
                actionB.SetTitle ("Click Me", UIControlState.Normal);
                Console.WriteLine ("adding target.");
                actionB.AddTarget (Handler, UIControlEvent.TouchUpInside);
                Console.WriteLine ("added target.");
                feedbackB = new UIButton (new CGRect (0, 200, 300, 150));
                feedbackB.SetTitle ("0", UIControlState.Normal);
                vc.View!.AddSubview (actionB);
                vc.View!.AddSubview (feedbackB);

                Window.RootViewController = vc;

                // make the window visible
                Window.MakeKeyAndVisible ();

                return true;
        }
```
on an iOS simulator I verified that (1) the code path in UIControl was getting hit and (2) that calling code worked as expected.

This is just the start - since everything is going to be built on top of this, I want this PR to get a lot of scrutiny.
The planned next steps are:
- change the generator to generate static byte arrays instead of strings
- change the existing manual bindings to use static byte arrays
- consider refactoring Selector instances to not use strings - this might not be a huge win. Don't know.
- when C# 11 is available, use the string form `"someString\0" u8` instead of byte arrays
- add an `Export8` or `ExportAscii` attribute that takes a byte array and refactor to use that

Ref: #10462.

### CODE REVIEW

**Review by rolfbjarne — CHANGES_REQUESTED** [review]
There's a failing test (that assumes there's only one GetHandle method):

https://github.com/xamarin/xamarin-macios/blob/37b8a0709caa4248cdb2db2fc020f45cc2db2d64/tests/linker/ios/link%20all/PreserveTest.cs#L155-L156

A more complicated matter is that the Selector.GetHandle call is optimized in the AOT compiler: https://github.com/dotnet/runtime/blob/f58cee20c39591835a01c2ffd5f2b3900b279c38/src/mono/mono/mini/intrinsics.c#L1941-L1981, I'm unsure how adding another overload will affect AOT compilation (nor how it will affect both speed and size). Some testing here would be nice.

File: src/UIKit/UIControl.cs [hunk 1]
```diff
@@ -22,6 +22,11 @@ class UIControlEventProxy : NSObject {
 		UIControl source;
 		internal int Counter = 1;
 		internal const string BridgeSelector = "BridgeSelector";
+		internal static byte [] BridgeSelector8 = new byte [] {
+			(byte)'B', (byte)'r', (byte)'i', (byte)'d', (byte)'g', (byte)'e', (byte)'S', (byte)'e',
```
Comment (Therzok) [inline src/UIKit/UIControl.cs:26]: C# 11 has UTF8 string literals:
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-11.0/utf8-string-literals

But the docs also details a way a different way to do that (that uses embedded assembly metadata).

Comment (Therzok) [reply src/UIKit/UIControl.cs:26]: Seems like that example is outdated.

https://devblogs.microsoft.com/dotnet/csharp-11-preview-updates/#utf-8-string-literals

Comment (stephen-hawley) [reply src/UIKit/UIControl.cs:26]: Right, can't wait to get access to that. In the meantime, it's byte arrays.

**Pull request closed without merging**